uaclient/apt.py

```diff
@@ -22,36 +23,37 @@
 CA_CERTIFICATES_FILE = '/usr/sbin/update-ca-certificates'
 
 
-class InvalidAPTCredentialsError(RuntimeError):
```

Comment:
I'm uncertain whether we want specialized exceptions for InvalidAptCredentials vs AptRepoTimeout, or just raising UserFacingError with the proper error message. Since we don't really react differently to a raised InvalidAptCreds vs Timeout, I went with using UserFacingError with the proper error message.

Reply:
I think there are a couple of cases where we would want specialised exceptions: when we want to be able to catch a particular error (and so need a distinct class for the `except` clause), and when we want to produce the same error in multiple places (and want to ensure that the text remains consistent across all error sites).

I don't believe either of those apply here, so +1 on just `UserFacingError`.

Reply:
OK, having read through more carefully, I think we may actually want something like a `RetryableUserFacingError`, which the subp retrying logic will retry on (but which will still output nicer user output by dint of being a subclass of `UserFacingError`).

(I still don't think we need an exception per-message here, because we're only using each one once.)